app/Http/Controllers/Frontend/SettingsController.php

```diff
@@ -48,17 +48,28 @@ public function updateMainSettings(Request $request): RedirectResponse {
     public function updatePrivacySettings(Request $request): RedirectResponse {
         $validated = $request->validate([
                                             'private_profile'           => ['nullable'],
+                                            'follower_approval'         => ['nullable'],
                                             'prevent_index'             => ['required', 'gte:0', 'lte:1'],
                                             'default_status_visibility' => [
                                                 'required',
                                                 Rule::in(StatusVisibility::getList())
                                             ]
                                         ]);
 
+        $privateProfile   = isset($validated['private_profile']) && $validated['private_profile'] == 'on';
+        $followerApproval = isset($validated['follower_approval']) && $validated['follower_approval'] == 'on';
+
+        if ($privateProfile) {
+            $followerApproval = true;
+        }
+        if (!$followerApproval) {
+            $privateProfile = false;
```

Comment:
I'm not sure if that's a good idea. Wouldn't that mean that a private profile would become public if you uncheck the followerApproval?